### PR TITLE
[wasm] Process LinkerArg for wasm targets

### PR DESCRIPTION
**Pull request by lewing**
Putting this up for discussion rather than a final form, I think it makes sense to use common names

### CODE REVIEW

**Comment by SingleAccretion** [comment]
`CustomLinkerArg` is an internal property in the NAOT target. `LinkerArg`is it's public counterpart.

**Comment by lewing** [comment]
> `CustomLinkerArg` is an internal property in the NAOT targets. `LinkerArg` is its public counterpart.

I was working off the assumption that the use of CustomLinkerArg in wit-bindgen was correct, should that be changed to LinkerArg?

**Comment by SingleAccretion** [comment]
Yes.

**Comment by lewing** [comment]
It looks like only iOS consumes LinkerArgs right now @maraf can you take a look across the platforms?

**Comment by maraf** [comment]
The `CustomLinkerArg` is use exclusively in NativeAOT. The `LinkerArg` is used in Apple, Android and LibraryMode and is added to `CustomLinkerArg` collection in NativeAOT https://github.com/dotnet/runtime/blob/main/src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.targets#L352

File: src/mono/wasi/build/WasiApp.targets [hunk 1]
```diff
@@ -354,6 +354,7 @@
       otherwise it will be included in the dotnet.wasm and required -S http argument to wasmtime
       -->
       <_WasiLinkStepArgs Include="-Wl,--component-type,&quot;$([MSBuild]::NormalizePath('$(MicrosoftNetCoreAppRuntimePackRidNativeDir)', 'WasiHttpWorld_component_type.wit').Replace('\','/'))&quot;"/>
+      <_WasiLinkStepArgs Include="@(LinkerArg->'%(Identity)')"/>
```
Comment (lewing) [inline src/mono/wasi/build/WasiApp.targets:357]:
```suggestion
      <_WasiLinkStepArgs Include="@(LinkerArg)')"/>
```

**Comment by lewing** [comment]
> The `CustomLinkerArg` is use exclusively in NativeAOT. The `LinkerArg` is used in Apple, Android and LibraryMode and is added to `CustomLinkerArg` collection in NativeAOT https://github.com/dotnet/runtime/blob/main/src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.targets#L352

I just meant I missed the android use in a quick scan, thanks for the full reply.  This patch just sticks it in wasi targets, it should really be in the shared bits.

**Review by radekdoulik — APPROVED** [review]

**Review by maraf — APPROVED** [review]

**Pull request merged**